### ISSUE TITLE
Convert error in CartesianProductArray

### ISSUE THREAD

**Issue by mforets** [JuliaReach/Reachability.jl#55]
```julia
julia> A = randn(4, 4); X0 = BallInf(ones(4), 0.1)
LazySets.BallInf{Float64}([1.0, 1.0, 1.0, 1.0], 0.1)

julia> s = solve(ContinuousSystem(A, X0), :T=>0.1, :blocks=>[1,2]);
ERROR: MethodError: Cannot `convert` an object of type Array{LazySets.HPolygon,1} to an object of type LazySets.CartesianProductArray
This may have arisen from a call to the constructor LazySets.CartesianProductArray(...),
since type constructors fall back to convert methods.
Stacktrace:
 [1] LazySets.CartesianProductArray(::Array{LazySets.HPolygon,1}) at ./sysimg.jl:77
 [2] reach_explicit_blocks!(::Array{Float64,2}, ::Array{LazySets.HPolygon,1}, ::Reachability.Systems.ConstantNonDeterministicInput, ::Int64, ::Int64, ::Int64, ::Reachability.ReachSets.##21#23, ::Array{Int64,1}, ::A
rray{LazySets.CartesianProductArray,1}) at /Users/forets/.julia/v0.6/Reachability/src/ReachSets/reach_explicit_blocks.jl:151
 [3] #reach#19(::String, ::Float64, ::Bool, ::Bool, ::Bool, ::Bool, ::Array{Any,1}, ::Function, ::Reachability.Systems.DiscreteSystem, ::Int64) at /Users/forets/.julia/v0.6/Reachability/src/ReachSets/reach.jl:134
 [4] (::Reachability.ReachSets.#kw##reach)(::Array{Any,1}, ::Reachability.ReachSets.#reach, ::Reachability.Systems.DiscreteSystem, ::Int64) at ./<missing>:0
 [5] solve(::Reachability.Systems.ContinuousSystem, ::Reachability.Options) at /Users/forets/.julia/v0.6/Reachability/src/solve.jl:127
 [6] solve(::Reachability.Systems.ContinuousSystem, ::Pair{Symbol,Float64}, ::Pair{Symbol,Array{Int64,1}}, ::Vararg{Pair{Symbol,Array{Int64,1}},N} where N) at /Users/forets/.julia/v0.6/Reachability/src/solve.jl:192
```

one workaround is to do: `res[1] = CartesianProductArray(Xhat0) -> res[1] = CartesianProductArray{Float64, HPolygon{Float64}(Xhat0)`. otherwise we can change the input type to `Xhat0::Vector{HPolygon{Float64}}`.

**Comment by mforets** [comment]
ok i think that the second one is preferable and it amounts to updating [the initial set convert line](https://github.com/JuliaReach/Reachability.jl/blob/master/src/ReachSets/reach.jl#L75).

**Comment by schillic** [comment]
This will be resolved by #43.
